### PR TITLE
Add provider & new stack labels to label sync, remove catalog repo from github config

### LINKED ISSUE #1194: Codify new stack labels in labels.yml
## Description

At some point recently we added the following stack labels:

- [🧱 stack: documentation](https://github.com/WordPress/openverse/issues?q=is%3Aissue+is%3Aopen+label+label%3A%22%F0%9F%A7%B1+stack%3A+documentation%22)
- [🧱 stack: infra](https://github.com/WordPress/openverse/issues?q=is%3Aissue+is%3Aopen+label+label%3A%22%F0%9F%A7%B1+stack%3A+infra%22)

These are not currently codified in our `labels.yml` file. It would be useful for them to be defined there so they can be shared across our repos (and be used as a source of truth):

https://github.com/WordPress/openverse/blob/4cb78373557c4619ec67fb2ac570d2e28d9e0994/automations/data/labels.yml#L85-L102

## Additional context

See #885 for an example

### PR DESCRIPTION
**Pull request by AetherUnbound**
## Fixes

Fixes #1194 by @AetherUnbound 

## Description

This PR makes a few changes:
* Adds the `☁️  provider` labels to the set of defined labels
* Adds the new stack labels to the defined set as well
* Removes `openverse-catalog` from `github.yml`, prevent it from being accessed as part of various automations

## Testing Instructions

I've already run the sync script off this branch, check out the labels here! https://github.com/WordPress/openverse/labels?q=provider

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by zackkrida — COMMENTED** [review]
Nice!

**Review by zackkrida — APPROVED** [review]

File: automations/data/labels.yml [hunk 1]
```diff
@@ -180,6 +184,24 @@ groups:
...
+      - name: audio
+        description: Audio provider APIs
+      - name: images
+        description: Image provider APIs
+      - name: text
+        description: Text provider APIs
+      - name: video
+        description: Video provider APIs
+      - name: 3D models
+        description: 3D model provider APIs
```
Comment (obulat) [inline automations/data/labels.yml:203]: I think we should remove "APIs" to account for other ways of ingestion (like iNaturalist).

**Review by obulat — APPROVED** [review]
Nice! A non-blocking comment inline

**Review by dhruvkb — APPROVED** [review]
While I personally see only limited utility of the `labels.yml` file now (and even less in the future once we figure out how to merge the infra into this repo), these are good changes.

LGTM!

**Comment by AetherUnbound** [comment]
> While I personally see only limited utility of the `labels.yml` file now (and even less in the future once we figure out how to merge the infra into this repo), these are good changes.

We're also using `labels.yml` for things like the automatic PR labelling, so I think it's worth having around: https://github.com/WordPress/openverse/blob/184cac6c446ccee5224f27d3f09dba6f73c2a680/automations/python/label_pr.py

It's nice to have a single source of truth for various labels, especially as we build out more automation with the new project boards around this IMO!

**Pull request merged**